src/auth/ui/vue/show.ts

```diff
@@ -332,18 +351,382 @@ export class AuthWebview extends VueWebview {
+        this.emitAuthAttempt({
+            authType: args.authType,
+            featureType: args.featureType,
+            result: 'Succeeded',
+            attempts: authAttempts,
+        })
+    }
+
+    #totalAuthAttempts: number = 0
+
+    /**
+     * This metric is emitted on an attempt to signin/connect/submit auth regardless
+     * of success.
+     */
+    private async emitAuthAttempt(args: {
+        authType: CredentialSourceId
+        featureType: FeatureId
+        result: Result
+        reason?: string
+        invalidFields?: string[]
+        attempts: number
+    }) {
+        telemetry.auth_addConnection.emit({
+            source: this.#authSource ?? '',
+            credentialSourceId: args.authType,
+            featureId: args.featureType,
+            result: args.result,
+            reason: args.reason,
+            invalidInputFields: args.invalidFields ? builderCommaDelimitedString(args.invalidFields) : undefined,
+            attempts: args.attempts,
+            isAggregated: true,
+        })
+
+        this.#totalAuthAttempts += args.attempts
+
+        if (args.result === 'Succeeded') {
+            // Clear the variables that track the previous uncompleted auth form
+            // since this was successfully completed.
+            this.previousAuthType = undefined
+            this.previousFeatureType = undefined
+            this.#previousFailedAuth = undefined
+        }
+    }
+
+    /**
+     * The metric emitted when the webview is closed by the user.
+     */
+    async emitWebviewClosed() {
+        if (this.previousFeatureType && this.previousAuthType) {
+            // We are closing the webview, and have an auth form if they were
+            // interacting but did not complete it.
+            await this.stopAuthFormInteraction(this.previousFeatureType, this.previousAuthType)
+        }
+
+        const allConnectedAuths = this.getAllConnectedAuths()
+        const newConnectedAuths = new Set(
+            [...allConnectedAuths].filter(value => !this.initialConnectedAuths.has(value))
+        )
+        const uncountedBuilderIds = this.getUncountedBuilderIds(allConnectedAuths, newConnectedAuths)
+
+        let numAuthConnections = (await this.getConnectionCount()) + uncountedBuilderIds
+        let numNewAuthConnections = numAuthConnections - this.initialNumConnections! + uncountedBuilderIds
+
+        if (numNewAuthConnections < 0) {
+            // Edge Case:
+            // numAuthConnections gets the latest number of connections, and it is
+            // possible that the user signed out or removed connections they initially had.
+            // If this is the case, we will set the total connections to what it initially was
+            // since we don't consider removing connections as not having existed.
+            numAuthConnections = this.initialNumConnections!
+            numNewAuthConnections = newConnectedAuths.size // best effort guess but can be wrong
+        }
```

Comment:
I don't think it would be unreasonable to leave it as a negative number. They removed a connection, so the # of new connections is negative.